packages/fx-core/src/component/driver/file/updateLaunchUrlInLaunchSettings.ts

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as fs from "fs-extra";
+import { Service } from "typedi";
+
+import { hooks } from "@feathersjs/hooks/lib";
+import { FxError, Result, SystemError, UserError } from "@microsoft/teamsfx-api";
+
+import { getLocalizedString } from "../../../common/localizeUtils";
+import { getAbsolutePath, wrapRun } from "../../utils/common";
+import { logMessageKeys } from "../aad/utility/constants";
+import { DriverContext } from "../interface/commonArgs";
+import { ExecutionResult, StepDriver } from "../interface/stepDriver";
+import { addStartAndEndTelemetry } from "../middleware/addStartAndEndTelemetry";
+import { updateProgress } from "../middleware/updateProgress";
+import { InvalidActionInputError, UnhandledError } from "../../../error/common";
+import { UpdateLaunchUrlInLaunchSettingsArgs } from "./interface/UpdateLaunchUrlInLaunchSettingsArgs";
+import { AppStudioScopes } from "../../../common/tools";
+
+const actionName = "file/updateLaunchUrlInLaunchSettings";
```

Comment:
To be more general, how about `file/createOrUpdateLaunchProfile`?
Not just text replacement, but to modify the whole launch profile.
E.g., input *name, hub, appid, ...*, and output the whole launch profile.

Reply:
In vs 'launchSettings.json', we have comment content. It is not easy to handle this part like json content. So, I just replace 'launchUrl' using regex. Or should I just use another style to generate and replace whole 'launchSettings.json' using an action?

Reply:
I remember we already have such json comment logic (v5 tasks.json migration?).

Reply:
BTW, in the VS schema (https://json.schemastore.org/launchsettings.json) it seems to be called *Debug Profile*

Reply:
Got it, I will update this part using comment json.

Reply:
Done.